test/PoolManager.t.sol

```diff
@@ -738,6 +794,224 @@ contract PoolManagerTest is Test, Deployers, GasSnapshot {
+                    | Hooks.BEFORE_SWAP_RETURNS_DELTA_FLAG
+            )
+        );
+        CustomCurveHook impl = new CustomCurveHook(manager);
+        vm.etch(hookAddr, address(impl).code);
+
+        (key,) = initPool(currency0, currency1, IHooks(hookAddr), 100, SQRT_RATIO_1_1, ZERO_BYTES);
+        // add liquidity by sending tokens straight into the contract
+        key.currency0.transfer(hookAddr, 10e18);
+        key.currency1.transfer(hookAddr, 10e18);
+
+        uint256 balanceBefore0 = currency0.balanceOf(address(this));
+        uint256 balanceBefore1 = currency1.balanceOf(address(this));
+
+        uint256 amountToSwap = 123456;
+        PoolSwapTest.TestSettings memory testSettings =
+            PoolSwapTest.TestSettings({takeClaims: false, settleUsingBurn: false});
+        IPoolManager.SwapParams memory params = IPoolManager.SwapParams({
+            zeroForOne: true,
+            amountSpecified: -int256(amountToSwap),
+            sqrtPriceLimitX96: SQRT_RATIO_1_2
+        });
+        swapRouter.swap(key, params, testSettings, ZERO_BYTES);
+
+        // the custom curve hook is 1-1 linear
+        assertEq(currency0.balanceOf(address(this)), balanceBefore0 - amountToSwap, "amount 0");
+        assertEq(currency1.balanceOf(address(this)), balanceBefore1 + amountToSwap, "amount 1");
+    }
+
+    function test_swap_beforeSwapNoOpsSwap_exactOutput() public {
+        address hookAddr = address(
+            uint160(
+                Hooks.AFTER_SWAP_FLAG | Hooks.AFTER_SWAP_RETURNS_DELTA_FLAG | Hooks.BEFORE_SWAP_FLAG
+                    | Hooks.BEFORE_SWAP_RETURNS_DELTA_FLAG
+            )
+        );
+        CustomCurveHook impl = new CustomCurveHook(manager);
+        vm.etch(hookAddr, address(impl).code);
+
+        (key,) = initPool(currency0, currency1, IHooks(hookAddr), 100, SQRT_RATIO_1_1, ZERO_BYTES);
+        // add liquidity by sending tokens straight into the contract
+        key.currency0.transfer(hookAddr, 10e18);
+        key.currency1.transfer(hookAddr, 10e18);
+
+        uint256 balanceBefore0 = currency0.balanceOf(address(this));
+        uint256 balanceBefore1 = currency1.balanceOf(address(this));
+
+        uint256 amountToSwap = 123456;
+        PoolSwapTest.TestSettings memory testSettings =
+            PoolSwapTest.TestSettings({takeClaims: false, settleUsingBurn: false});
+        IPoolManager.SwapParams memory params = IPoolManager.SwapParams({
+            zeroForOne: true,
+            amountSpecified: int256(amountToSwap),
+            sqrtPriceLimitX96: SQRT_RATIO_1_2
+        });
+        swapRouter.swap(key, params, testSettings, ZERO_BYTES);
+
+        // the custom curve hook is 1-1 linear
+        assertEq(currency0.balanceOf(address(this)), balanceBefore0 - amountToSwap, "amount 0");
+        assertEq(currency1.balanceOf(address(this)), balanceBefore1 + amountToSwap, "amount 1");
+    }
+
+    int128 constant maxPossibleIn = -6018336102428409;
```

Comment:
are these constants for a specific pool w specific amount of liquidity?

Reply:
in that case maybe should not be constants and just move them down to the specific test?